apps/web-client/README.md

```diff
@@ -1,38 +1,282 @@
+
+The data is retrieved from the db, primarily through the db streams (with respect to the view).
+
+The data streams are transformed in `lib/stores`, combining/transforming the data, and made available to the UI as svelte stores.
+
+##### Binding the element value to svelte store
+
+To display/update the data streamed, we simply bind the stores to interactive elements, allowing for bidirectional updates.
+
+The way this works, under the hood, is:
+
+-   Svelte performs `store.subscribe` to pass the value to the element, in diagrams we'll call this `bind:subscribe`
+-   each time the element gets updated (sort of `on:change`), Svelte performs `store.set` with the updated value, in diagrams, we'll call this `bind:set`
+-   when the element is destroyed, the store gets unsubscribed from
+
+##### 'displayName' store
+
+The `displayName` store streams the display name from the db entry (note/warehouse) and provides a `set` method used to update the same displayName (in the db):
+
+```mermaid
+graph LR;
+  element -->|bind:set| store -->|"entity.setName()"| entity
+  entity["note/warehouse document"] -->|"entity.stream().displayName"| store["displayName store"]
+  store -->|bind:subscribe| element["display name element"]
+```
+
+To create a display name store, we use the factory function:
+
+```typescript
+const warehouse = db.warehouse('warehouse-1');
+const note = wareshouse.note('note-1');
+
+const wDisplayName = createDisplayNameStore(warehouse, null, {});
+const nDisplayName = createDisplayNameStore(note, null, {});
+```
+
+The second param (`null` in the example) is used to pass in the internal state store, more on that below.
+
+##### 'state' store
+
+The state store connects the note state displayed in the UI with the one in the db.
+
+Since the current implementation doesn't save the note state the same way it's displayed in the UI, we have to transform the value going back and forth.
+
+Note states displayed by the UI are as follows: `Draft`, `Committed`, `Deleted`, whereas the state in the note document, in the db, is controlled by the `committed: true | false` property.
+
+| UI        | note (in db)                                              |
+| --------- | --------------------------------------------------------- |
+| Draft     | committed: `false \| undefined`                           |
+| Committed | committed: `true`                                         |
+| Deleted   | doesn't exist in the db (in the future this might change) |
+
+Additionally, in the UI, we're using temp states, one for each state displayed in the UI:
+
+| UI state  | UI temp state |
+| --------- | ------------- |
+| Draft     | Saving        |
+| Committed | Committing    |
+| Deleted   | Deleting      |
+
+The way temp states work:
+
+-   if we're updating the draft note (this is normally an outside update), the state displayed is "Saving", then when the note gets updated, the note's current state in db (Draft) is streamed back to the display store
```

Comment:
Is this meant to be state in db (committed : false)? since "draft" is not a db state

Reply:
Yes...here I meant the state from the db that gets streamed back (not the exact implementation). This is not an error, but I'll revisit if it should be explained better.